python/cutlass_library/generator.py

```diff
@@ -2225,9 +2225,13 @@ def GenerateSM80_TensorOp_16816_mixed_input_upcast_a(manifest, cuda_version):
       math_inst.element_accumulator,
     ]
 
-    CreateGemmOperator(manifest, layouts, tile_descriptions, \
+    operations = CreateGemmOperator(manifest, layouts, tile_descriptions, \
       data_type, alignment_constraints)
 
+    for op in operations:
+      if (data_type[2] == DataType.f16) and (op.tile_description.threadblock_shape[1] <= 32):
```

Comment:
Suggestions:

- Using `op` instead of data_type for consistency with `op.tile_description`
- Using DataTypeSize to cover f16 and bf16.

```cpp
if DataTypeSize[op.C.element] == 16 and (op.tile_description.threadblock_shape[1] <= 32) :
```

- `Use operation  +=  CreateGemmOperator(manifest, layouts, tile_descriptions` on line 2245 and move the for loop out of `if` block to combine the setting alignment for `F32` and `F16` accumulation at one place? Will that work.

Reply:
do you want to take this over?

Reply:
Made the change but cannot push to NVIDIA/cutlass.git branch `fix/mixed_h16816_alignc`

```
ERROR: Permission to NVIDIA/cutlass.git denied to manishucsd.
fatal: Could not read from remote repository.
```

Please find the change below:

```
(fix/mixed_h16816_alignc) $ git diff
diff --git a/python/cutlass_library/generator.py b/python/cutlass_library/generator.py
index 89fe9856..dbc0cde1 100644
--- a/python/cutlass_library/generator.py
+++ b/python/cutlass_library/generator.py
@@ -2228,10 +2228,6 @@ def GenerateSM80_TensorOp_16816_mixed_input_upcast_a(manifest, cuda_version):
     operations = CreateGemmOperator(manifest, layouts, tile_descriptions, \
       data_type, alignment_constraints)
 
-    for op in operations:
-      if (data_type[2] == DataType.f16) and (op.tile_description.threadblock_shape[1] <= 32):
-        op.C.alignment = 4
-
     # Avoid emitting two kernels if the accumulator type does not differ from the input type (e.g. F16 accumulation)
     if math_inst.element_a != math_inst.element_accumulator:
 
@@ -2242,12 +2238,13 @@ def GenerateSM80_TensorOp_16816_mixed_input_upcast_a(manifest, cuda_version):
         math_inst.element_accumulator,
       ]
 
-      operations = CreateGemmOperator(manifest, layouts, tile_descriptions, \
+      operations += CreateGemmOperator(manifest, layouts, tile_descriptions, \
         data_type_mixed, alignment_constraints) 
     
-      for op in operations:
-        if op.tile_description.threadblock_shape[1] <= 32:
-          op.C.alignment = 4
+    for op in operations:
+      if (DataTypeSize[op.C.element] == 16) and \
+         (op.tile_description.threadblock_shape[1] <= 32):
+        op.C.alignment = 4
```